### PR TITLE
Beginning C++23: Chapter 15

### PR DESCRIPTION
**Pull request by Peter-VanWeert**

### CODE REVIEW

File: Examples/Modules/Chapter 15/Ex15_06/ToughPack.cppm [hunk 1]
```diff
@@ -9,7 +9,6 @@ public:
   using Box::Box;
 
 protected:
-  // Function to calculate volume of a ToughPack allowing 15% for packing
-  double volume() const override { return 0.85 * m_length * m_width * m_height; }
+  // Function to calculate volume of a ToughPack allowing 13% for packing
+  double volume() const override { return 0.87 * getLength() * getWidth() * getHeight(); }
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 15/Ex15_06/ToughPack.cppm:13]: This is still protected. On purpose?

File: Examples/Modules/Chapter 15/Ex15_09/ToughPack.cppm [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export module boxes:tough_pack;
+import :box;
+import <iostream>;
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 15/Ex15_09/ToughPack.cppm:3]: should be: `import <print>;`

File: Examples/Modules/Chapter 15/Ex15_09/ToughPack.cppm [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+export class ToughPack : public Box
+{
+public:
+  ToughPack(double length, double width, double height)
+    : Box{length, width, height}
+  {
+    std::println("ToughPack constructor called for a Box of volume {}", volume());
+  }
+  virtual ~ToughPack()
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 15/Ex15_09/ToughPack.cppm:13]: The text says: `~ToughPack() override`

File: Examples/Modules/Chapter 15/Ex15_09/Ex14_09.cpp [hunk 4]
```diff
@@ -4,5 +4,5 @@ import boxes;
 int main()
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 15/Ex15_09/Ex14_09.cpp:4]: This file should be called `Ex15_09.cpp`

File: Examples/Modules/Chapter 15/Ex15_09/Ex14_09.cpp [hunk 5]
```diff
@@ -4,5 +4,5 @@ import boxes;
...
 {
   ToughPack toughPack{ 1.0, 2.0, 3.0 };
-  toughPack.showVolume();     // Should show a volume equal to 85% of 1x2x3, or 5.1
+  toughPack.printVolume();   // Should print a volume equal to 87% of 1x2x3, or 5.1
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 15/Ex15_09/Ex14_09.cpp:7]: 5.22 instead of 5.1

File: Examples/Modules/Chapter 15/Ex15_12/Ex15_11.cpp [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// Using an interface class and indirect base classes
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 15/Ex15_12/Ex15_11.cpp:1]: This file should be called `Ex15_12.cpp`

File: Examples/Modules/Chapter 15/Ex15_07/ToughPack.cppm [hunk 7]
```diff
@@ -1,17 +1,15 @@
...
   // Inherit the Box(length, width, height) constructor
   using Box::Box;
-  ~ToughPack() { std::cout << "ToughPack destructor called" << std::endl; }
+  ~ToughPack() { std::println("ToughPack destructor called"); }
 
 protected:
-  // Function to calculate volume of a ToughPack allowing 15% for packing
-  double volume() const override { return 0.85 * m_length * m_width * m_height; }
+  // Function to calculate volume of a ToughPack allowing 13% for packing
+  double volume() const override { return 0.87 * getLength() * getWidth() * getHeight(); }
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 15/Ex15_07/ToughPack.cppm:14]: This is still protected. On purpose?

**Review by MarcGregoire — COMMENTED** [review]
Done with review.

File: Examples/Modules/Chapter 15/Ex15_09/ToughPack.cppm (hunk 2 above)
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 15/Ex15_09/ToughPack.cppm]: Made it `import std;` instead, as it should be (using the usual trick to make it compile).

**Pull request merged**